ocaml/xcp-rrdd/bin/rrdd/rrdd_server.ml

```diff
@@ -715,9 +715,25 @@ module Plugin = struct
          specified unique ID. If the plugin is not registered, -1 is returned. *)
       let next_reading (uid : P.uid) : float =
         let open Rrdd_shared in
+        let ( --- ) a b = Mtime.Span.abs_diff a b in
         if with_lock registered_m (fun _ -> Hashtbl.mem registered uid) then
-          with_lock last_loop_end_time_m (fun _ ->
-              !last_loop_end_time +. !timeslice -. Unix.gettimeofday ()
+          with_lock next_iteration_start_m (fun _ ->
+              let current = Mtime_clock.count from_loop_start in
+              let diff =
+                Mtime.Span.to_float_ns
+                  (!Rrdd_shared.next_iteration_start
+                  --- Mtime_clock.count from_loop_start
+                  )
+                /. 1_000_000_000.
```

Comment:
Why not import `clock` and use `[Clock.Timer.span_to_s](https://github.com/xapi-project/xen-api/blob/master/ocaml/libs/clock/timer.mli#L49)`?